addons/mail/static/src/core/common/thread_model.js

```diff
@@ -696,6 +696,10 @@ export class Thread extends Record {
         if (!thread) {
             return;
         }
+        if (this.self_member_id && !this.self_member_id.is_pinned) {
+            this.self_member_id.is_pinned = true;
+            this.channel.pinRpc({ pinned: true });
+        }
```

Comment:
All of those open methods are confusing, but I'm pretty sure this can miss flows.

I'm not sure what is the best place, but I'm thinking about onAdd of the chatWindow on channel.